cookbook/src/tx_segwit-v0.md

```diff
@@ -0,0 +1,342 @@
+We are using the `new` method to instantiate the struct with the `unsigned_tx` that we defined earlier.
+`new` takes any `Borrow<Transaction>` as an argument.
+[`Borrow<T>`](https://doc.rust-lang.org/std/borrow/trait.Borrow.html) is a trait that allows us to pass either a reference to a `T` or a `T` itself.
+Hence, you can pass a `Transaction` or a `&Transaction` to `new`.
+
+`sighash_cache` is instantiated as mutable because we will be updating it by adding a [`segwit_signature_hash`](https://docs.rs/bitcoin/latest/bitcoin/sighash/struct.SighashCache.html#method.segwit_signature_hash) to it.
+This computes the [BIP143](https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki) sighash for any flag type.
+It takes the following arguments:
+
+- `input_index` is the index of the input we are signing; it is a [`usize`](https://doc.rust-lang.org/std/primitive.usize.html) type.
+   We are using `0` since we only have one input.
+- `script_code` is the script code required to spend a P2WPKH output; it is a reference to [`Script`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/script/struct.Script.html) type.
+   We are using the `script_code` variable that we defined earlier.
+- `value` is the amount of the UTXO we are spending; it is a [`u64`](https://doc.rust-lang.org/std/primitive.u64.html) type.
+   We are using the `const DUMMY_UTXO_AMOUNT` that we defined earlier.
+- `sighash_type` is the type of sighash; it is a [`EcdsaSighashType`](https://docs.rs/bitcoin/latest/bitcoin/sighash/enum.EcdsaSighashType.html) enum.
+   We are using the [`All`](https://docs.rs/bitcoin/latest/bitcoin/sighash/enum.EcdsaSighashType.html#variant.All) variant,
+   that sign all the outputs.
+
+We compute the message `msg` by converting the `sighash` to a [`Message`](https://docs.rs/secp256k1/latest/secp256k1/struct.Message.html) type.
+This is a hashed message input to an ECDSA signature.
+The [Message::from](https://docs.rs/secp256k1/latest/secp256k1/struct.Message.html#impl-From%3C%26%27_%20bitcoin%3A%3Ahashes%3A%3Asha256d%3A%3AHash%3E) method takes anything that implements the [`Hash` trait](https://docs.rs/bitcoin_hashes/latest/bitcoin_hashes/trait.Hash.html) as an argument[^ThirtyTwoByteHash].
+
+We compute the signature `sig` by using the [`sign_ecdsa`](https://docs.rs/secp256k1/latest/secp256k1/struct.Secp256k1.html#method.sign_ecdsa) method.
+It takes a refence to a [`Message`](https://docs.rs/secp256k1/latest/secp256k1/struct.Message.html) and a reference to a [`SecretKey`](https://docs.rs/secp256k1/latest/secp256k1/struct.SecretKey.html) as arguments,
+and returns a [`Signature`](https://docs.rs/secp256k1/latest/secp256k1/ecdsa/struct.Signature.html) type.
+
+In the next step, we update the `sighash_cache` first with [`witness_mut`](https://docs.rs/bitcoin/latest/bitcoin/sighash/struct.SighashCache.html#method.witness_mut) method.
```

Comment:
Perhaps?
```suggestion
In the next step, we update the witness stack for the input we just signed using the `SighashCache::witness_mut`(https://docs.rs/bitcoin/latest/bitcoin/sighash/struct.SighashCache.html#method.witness_mut) method.
```

Reply:
This would not render a markdown link, we are missing the brackets `[ ]`.